src/mbed_tools/targets/_internal/target_attributes.py

```diff
@@ -80,12 +80,13 @@ def _read_json_file(path_to_file: pathlib.Path) -> Any:
         FileNotFoundError: path provided does not lead to a valid json file
     """
     try:
-        return json.loads(path_to_file.read_text())
+        # mypy doesn't recognise that json.loads always returns a dictionary
+        return cast(dict, json.loads(path_to_file.read_text()))
     except JSONDecodeError as json_err:
```

Comment:
Are there potential exceptions from the cast we should handle here or should we just let the exception bubble up?

Reply:
At runtime `cast` is a no-op, it only executes any code when type checking